src/pages/index.tsx

```diff
@@ -1,29 +1,44 @@
 import CampaignHero from "../components/hero/campaignHero";
-import { featuredImageID } from "../utils/utils";
-import {
-  SkipNavigation,
-  TemplateAppContainer,
-} from "@nypl/design-system-react-components";
+import SwimLanes from "../components/swimlanes/swimLanes";
+import { TemplateAppContainer } from "@nypl/design-system-react-components";
+import data from "@/data/lanes";
+import { getNumItems, featuredImageID } from "@/utils/utils";
 
 export default function Home(props: any) {
   return (
     <TemplateAppContainer
       aboveHeader={<p> Notification banner </p>}
       header={<p> Header </p>}
       breakout={<CampaignHero featuredImageID={props.featuredImageID} />}
-      contentPrimary={
-        <>
-          <p>First swim lane</p>
-          <p>Featured Content</p>
-          <p>Rest of swim lanes</p>
-          <p>Explore further links</p>
-        </>
-      }
+      contentPrimary={<SwimLanes lanesWithNumItems={props.lanesWithNumItems} />}
       renderSkipNavigation={true}
     />
   );
 }
 
 export async function getServerSideProps() {
-  return { props: { featuredImageID: featuredImageID() } };
+  const lanes = data.lanes;
+  const lanesWithNumItems = await Promise.all(
+    lanes.map(async (lane) => {
+      const updatedCollections = await Promise.all(
```

Comment:
Look into `Promise.allSettled` and see if you can use it. And you may not need the outer `Promise.all`.